presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/KinesisConnectorFactory.java

```diff
@@ -0,0 +1,168 @@
+import io.airlift.json.JsonModule;
+import io.airlift.log.Logger;
+import io.prestosql.connector.ConnectorId;
+import io.prestosql.spi.NodeManager;
+import io.prestosql.spi.connector.Connector;
+import io.prestosql.spi.connector.ConnectorContext;
+import io.prestosql.spi.connector.ConnectorFactory;
+import io.prestosql.spi.connector.ConnectorHandleResolver;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.spi.type.TypeManager;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This factory class creates the KinesisConnector during server start and binds all the dependency
+ * by calling create() method.
+ */
+public class KinesisConnectorFactory
+        implements ConnectorFactory
+{
+    public static final String connectorName = "kinesis";
+    private static final Logger log = Logger.get(KinesisConnectorFactory.class);
+
+    private final ClassLoader classLoader;
+    private TypeManager typeManager;
+    private NodeManager nodeManager;
+    private Optional<Supplier<Map<SchemaTableName, KinesisStreamDescription>>> tableDescriptionSupplier = Optional.empty();
+    private Map<String, String> optionalConfig = ImmutableMap.of();
+    private Optional<Class<? extends KinesisClientProvider>> altProviderClass = Optional.empty();
+
+    private KinesisHandleResolver handleResolver;
+
+    private Injector injector;
+
+    KinesisConnectorFactory(ClassLoader classLoader, Optional<Supplier<Map<SchemaTableName, KinesisStreamDescription>>> tableDescriptionSupplier,
+            Map<String, String> optionalConfig,
+            Optional<Class<? extends KinesisClientProvider>> altProviderClass)
+    {
+        this.classLoader = classLoader;
+        this.tableDescriptionSupplier = requireNonNull(tableDescriptionSupplier, "tableDescriptionSupplier is null");
+        this.optionalConfig = requireNonNull(optionalConfig, "optionalConfig is null");
+        this.altProviderClass = requireNonNull(altProviderClass, "altProviderClass is null");
+
+        this.handleResolver = new KinesisHandleResolver(connectorName);
+
+        // Explanation: AWS uses a newer version of jackson (2.6.6) than airlift (2.4.4).  In order to upgrade
+        // to the latest version of the AWS API, we need to turn this feature off.  This can be set
+        // in jvm.properties but trying to make this more foolproof.
+        System.setProperty("com.amazonaws.sdk.disableCbor", "true");
+    }
+
+    @Override
+    public String getName()
+    {
+        return connectorName;
+    }
+
+    @Override
+    public ConnectorHandleResolver getHandleResolver()
+    {
+        return new KinesisHandleResolver(connectorName);
+    }
+
+    @Override
+    public Connector create(String connectorId, Map<String, String> config, ConnectorContext context)
+    {
+        log.info("In connector factory create method.  Connector id: " + connectorId);
```

Comment:
Please remove this. We don't add such debug level logging, especially at info level, since it doesn't provide any value to end users.

Reply:
removed